extensions/narayana-stm/runtime/src/main/java/io/quarkus/narayana/stm/runtime/ObjectStoreEnvironmentBeanSubstitution.java

```diff
@@ -0,0 +1,18 @@
+package io.quarkus.narayana.stm.runtime;
+
+import java.io.File;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+@TargetClass(className = "com.arjuna.ats.arjuna.common.ObjectStoreEnvironmentBean")
+public final class ObjectStoreEnvironmentBeanSubstitution {
+
+    /**
+     * @return fixed ObjectStore path resolved during runtime
+     */
+    @Substitute
+    public String getObjectStoreDir() {
+        return System.getProperty("user.home") + File.separator + "ObjectStore";
```

Comment:
Are we sure this won't be cache this at build time?

Reply:
Are you afraid that the `System.getProperty("user.home")` is cached at build time?